balance_check/validators/gift_card.py

```diff
@@ -0,0 +1,44 @@
+import re
+from enum import Enum
+from luhn import verify
+
+
+class Merchant(Enum):
+    GameStop = 'GameStop'
+
+
+merchant_regex = {
+    Merchant.GameStop: re.compile('636491[0-9]{13}')
```

Comment:
This should match against the full string, i.e. add `^` and `$` to the pattern like this:
```suggestion
    Merchant.GameStop: re.compile('^636491[0-9]{13}$')
```